pkg/kubelet/volume/cache/desired_state_of_world.go

```diff
@@ -203,6 +217,25 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 	return volumeName, nil
 }
 
+func (dsw *desiredStateOfWorld) MarkVolumesReportedInUse(
+	reportedVolumes []api.UniqueVolumeName) {
+	dsw.Lock()
+	defer dsw.Unlock()
+
+	reportedVolumesMap := make(
+		map[api.UniqueVolumeName]bool, len(reportedVolumes) /* capacity */)
+
+	for _, reportedVolume := range reportedVolumes {
+		reportedVolumesMap[reportedVolume] = true
+	}
+
+	for volumeName, volumeObj := range dsw.volumesToMount {
+		_, volumeReported := reportedVolumesMap[volumeName]
+		volumeObj.reportedInUse = volumeReported
+		dsw.volumesToMount[volumeName] = volumeObj
```

Comment:
For future work - this is why you should never have a map where the value type is a struct.  If forces you to copy the struct out and write it back just to update a field.  If the map value were a pointer, no problem.